spec/betagouvbot/mailer_spec.rb

```diff
@@ -2,65 +2,31 @@
 # frozen_string_literal: true
 
 RSpec.describe BetaGouvBot::Mailer do
-  let(:in_3w) do
-    instance_double(
-      'mail',
-      format: { personalizations: [to: ['email' => 'ann@email.coop']] }
-    )
-  end
-
-  let(:in_2w) do
-    instance_double(
-      'mail',
-      format: {
-        personalizations: [
-          to: [{ 'email' => 'ann@email.coop' }, { 'email' => 'hi@email.coop' }]
-        ]
-      }
-    )
-  end
-
-  let(:rules) do
-    {
-      1  => { mail: instance_double('mail') },
-      14 => { mail: in_2w },
-      21 => { mail: in_3w }
-    }
-  end
+  let(:horizons) { BetaGouvBot::NotificationRule.horizons }
+  let(:schedule) { BetaGouvBot::Anticipator.(authors, horizons, Date.today) }
 
-  let(:schedule) { BetaGouvBot::Anticipator.(authors, rules.keys, Date.today) }
-
-  before { allow(described_class).to receive(:post) }
+  before { allow(described_class).to receive(:email) }
 
   describe 'selecting recipients of emails' do
-    context 'when a member has an end date in three weeks' do
-      let(:authors) { [id: 'ann', fullname: 'Ann', end: (Date.today + 21).iso8601] }
-
-      it 'sends an email directly to the author' do
-        described_class.(schedule, rules).map(&:execute)
```

Comment:
@Morendil This is entanglement : what you're testing is not `Mailer`'s unitary behaviour but the relationship between `Mailer` and `MailAction`. `#post` should be defined in `MailAction`.